docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+
+Wildcards (`*`) are supported in names and endpoint definitions.
+
+== Security realms
+
+The security realm is a named security configuration. It's intended mainly
+for member configuration but with limited support on Java clients too.
+Security realms in current Hazelcast versions allow defining identity
+and authentication configurations.
+
+== Advanced security topics
+
+=== Auditlog
+
+Hazelcast Auditlog is a security feature that records and logs
+security-related events and operations within the Hazelcast cluster,
+providing increased visibility and traceability for monitoring and
+compliance purposes.
+
+[[secret-strength-policy]]
+=== Validating Secrets Using Strength Policy
+
+Hazelcast's configurable secrets strength policy allows users to
+customize the security level of sensitive data, like passwords and
+encryption keys, based on their requirements. This policy enables users
+to define these secrets' minimum length, character sets, and complexity,
+ensuring a higher level of protection against unauthorized
+access. By offering this flexibility, Hazelcast empowers organizations
+balance security and usability according to their unique needs.
+
+[source,bash]
+----
+java -Dhazelcast.security.secret.policy.min.length=10
+    -Dhazelcast.security.dictionary.policy.wordlist.path=/opt/wordlist.txt ...
+----
+
+=== FIPS 140-2
+
+Hazelcast is tested in FIPS 140-2 enabled environments to ensure it works
+appropriately using cryptographic modules complying
+with the U.S. Federal Information Processing Standard.
+
+=== Security debugging
+
+Use standard Java system properties:
+
+[source,bash]
+----
+java -Djava.security.debug=all \
+     -Djavax.net.debug=all ...
+----
+
+Increase the log level for`com.hazelcast.security`. Sample in Log4J2
+config:
+
+[source,xml]
+----
+<Configuration>
+    <Loggers>
+        <Logger name="com.hazelcast.security" level="ALL"/>
+    </Loggers>
+</Configuration>
+----
+
+== Customizing security by code
+
+Hazelcast supports several extension points in the security area. Users
+may provide their own Java implementation of Hazelcast SPIs to cover
+their security needs fully.
+
+[[jaas-authn]]
+=== JAAS authentication
+
+Users can provide their login modules to be executed during
+authentication. The simplest way is to extend `ClusterLoginModule` base
+class in Hazelcast.
+
+Hazelcast uses three custom `Principal` types. They should be added to JAAS
+Subject during the authentication in login modules:
+
+* `ClusterIdentityPrincipal` - authenticated name
+* `ClusterRolePrincipal` - assigned role name
+* `ClusterEndpointPrincipal` - remote IP address
+
+=== AuditLog SPI
+
+Auditlog SPI allows users to provide their `AuditlogService` interface
+implementation. Such a custom implementation can trigger additional
+events or write to external systems like a Syslog server.
+
+A sample implementation of the SPI is in the code samples:
+https://github.com/hazelcast/hazelcast-code-samples/tree/master/enterprise/auditlog
```

Comment:
Maybe copying an interesting part of code could be nice.

Reply:
As the implementation covers several interfaces it's safer to leave it out of this document.

The code sample itself has a README explaining the implementation.